app/src/molecules/DeckMap/index.tsx

```diff
@@ -1,55 +1,47 @@
-// @flow
 import * as React from 'react'
 import { connect } from 'react-redux'
 import { withRouter } from 'react-router-dom'
 import some from 'lodash/some'
 import map from 'lodash/map'
 import mapValues from 'lodash/mapValues'
-import { type DeckSlotId } from '@opentrons/shared-data'
-import type { ContextRouter } from 'react-router-dom'
 
 import { RobotWorkSpace, Module as ModuleItem } from '@opentrons/components'
-// $FlowFixMe(mc, 2021-03.15): ignore until TS conversion
 import { getDeckDefinitions } from '@opentrons/components/src/deck/getDeckDefinitions'
 
-import type { State, Dispatch } from '../../redux/types'
-import {
-  selectors as robotSelectors,
-  type Labware,
-  type SessionModule,
-} from '../../redux/robot'
+import { selectors as robotSelectors } from '../../redux/robot'
+
+import type { DeckSlot, DeckSlotId } from '@opentrons/shared-data'
+import type { RouteComponentProps } from 'react-router-dom'
+import type { State } from '../../redux/types'
+import type { Labware, SessionModule, Slot } from '../../redux/robot'
 
 import { getMatchedModules } from '../../redux/modules'
 
 import { LabwareItem } from './LabwareItem'
 
 export * from './LabwareItem'
 
-type OP = {|
-  ...ContextRouter,
-  modulesRequired?: boolean,
-  enableLabwareSelection?: boolean,
-  className?: string,
-|}
+interface OP extends RouteComponentProps<{ slot?: string }> {
+  modulesRequired?: boolean
+  enableLabwareSelection?: boolean
+  className?: string
+}
 
-type DP = {| dispatch: Dispatch |}
+interface SP {
+  labwareBySlot?: { [slot: string]: Labware[] | undefined }
+  modulesBySlot?: { [slot: string]: DisplayModule | undefined }
+  selectedSlot?: DeckSlotId | null
+  areTipracksConfirmed?: boolean
+}
 
-type DisplayModule = {|
-  ...$Exact<SessionModule>,
-  mode?: $PropertyType<React.ElementProps<typeof ModuleItem>, 'mode'>,
-  usbInfoString?: string,
-|}
+type ModuleDisplayMode = React.ComponentProps<typeof ModuleItem>['mode']
 
-type SP = {|
-  labwareBySlot?: { [DeckSlotId]: Array<Labware> },
-  modulesBySlot?: {
-    [DeckSlotId]: ?DisplayModule,
-  },
-  selectedSlot?: ?DeckSlotId,
-  areTipracksConfirmed?: boolean,
-|}
+interface DisplayModule extends SessionModule {
+  mode?: React.ComponentProps<typeof ModuleItem>['mode']
+  usbInfoString?: string
+}
 
-type Props = {| ...OP, ...SP, ...DP |}
+type Props = OP & SP
```

Comment:
We could use `interface` here if we wanted

```suggestion
interface Props extends OP, SP {}
```